### PR TITLE
Remove Cassandra certs from repo

### LINKED ISSUE #538: Remove checked in certs from the repo
_Originally posted by @benbromhead in https://github.com/shotover/shotover-proxy/issues/437#issuecomment-1043810289_

### PR DESCRIPTION
**Pull request by conorbros**
closes #538

### CODE REVIEW

**Comment by conorbros** [comment]
Alright, fixed the failing test. One thing to remember from when this PR is merged is that if a test relies on the current directory being something particular it needs to be marked with `#[serial]`.

File: test-helpers/src/cert.rs [hunk 1]
```diff
@@ -34,3 +37,17 @@ pub fn generate_test_certs(path: &Path) {
     .unwrap();
     std::fs::write(path.join("redis.key"), cert.serialize_private_key_pem()).unwrap();
 }
+
+pub fn generate_cassandra_test_certs() {
+    let current_path = current_dir().unwrap();
+    set_current_dir("example-configs/cassandra-tls/certs").unwrap();
+
+    if let Err(ErrorKind::NotFound) = Command::new("./gen_certs.sh")
```
Comment (rukai) [inline test-helpers/src/cert.rs:45]: Shouldnt we fail on all errors?

File: test-helpers/src/cert.rs [hunk 2]
```diff
@@ -34,3 +37,17 @@ pub fn generate_test_certs(path: &Path) {
     .unwrap();
     std::fs::write(path.join("redis.key"), cert.serialize_private_key_pem()).unwrap();
 }
+
+pub fn generate_cassandra_test_certs() {
+    let current_path = current_dir().unwrap();
+    set_current_dir("example-configs/cassandra-tls/certs").unwrap();
```
Comment (rukai) [inline test-helpers/src/cert.rs:43]: If we panic in the middle that will leave other tests with a different working directory right?
Can this be written such that it doesnt change the directory?
Maybe the script should be responsible for settings its own directory?

TBH I would rather we just rewrite the script entirely using rust, I would need some convincing that the existing `generate_test_certs` can not be adapted for cassandra.
That said this stuff can be a bit complicated so im willing to compromise here for the sake of moving things forward.

File: shotover-proxy/example-configs/cassandra-tls/certs/gen_certs.sh [hunk 3]
```diff
@@ -1,17 +1,63 @@
...
-
 # generate keystore
 openssl pkcs12 -export -out keystore.p12 -inkey localhost.key -in localhost.crt -passout pass:password
 keytool -importkeystore -destkeystore keystore.jks -srcstoretype PKCS12 -srckeystore keystore.p12 -deststorepass "password" -srcstorepass "password"
 
 # generate truststore
 openssl pkcs12 -export -out truststore.p12 -inkey localhost.key -in localhost.crt -passout pass:password
 keytool -importkeystore -destkeystore truststore.jks -srcstoretype PKCS12 -srckeystore truststore.p12 -deststorepass "password" -srcstorepass "password"
+
+sudo chown -R 1001 keystore.p12
```
Comment (rukai) [inline shotover-proxy/example-configs/cassandra-tls/certs/gen_certs.sh:60]: We should be able to just set the appropriate permission bits via chmod which wont need root.

**Review by rukai — APPROVED** [review]
Would love to see this using the cert generation rewritten in rust one day for maintainability and cross platform reasons but as is this PR is an improvement over the status quo.

LGTM

File: .gitignore [hunk 4]
```diff
@@ -7,6 +7,12 @@ tests/soaktest-*
 *.gz
 /.vscode
 /shotover-proxy/example-configs/redis-tls/certs
+/shotover-proxy/example-configs/cassandra-tls/certs/*.jks
+/shotover-proxy/example-configs/cassandra-tls/certs/*.p12
+/shotover-proxy/example-configs/cassandra-tls/certs/*.key
+/shotover-proxy/example-configs/cassandra-tls/certs/*.srl
+/shotover-proxy/example-configs/cassandra-tls/certs/*.csr
+/shotover-proxy/example-configs/cassandra-tls/certs/*.crt
```
Comment (XA21X) [inline .gitignore:15]: Any reason to use specific file extensions instead of excluding the whole folder like for Redis?

Doesn't matter, just curious.

Comment (XA21X) [reply .gitignore:15]: Nvm, I missed the fact that the script is in the same folder.

**Review by XA21X — APPROVED** [review]

**Pull request merged**